datamade/macoupin-directory
# upgrade to CARTO

This PR is a complete rewrite of this project using CARTO instead of Fusion Tables.

Before:

<img width="1276" alt="Screen Shot 2019-11-12 at 5 00 17 PM" src="https://user-images.githubusercontent.com/919583/68718221-3ad80b00-056e-11ea-9f5a-a4dacab2e024.png">

After:

<img width="1275" alt="Screen Shot 2019-11-14 at 4 06 34 PM" src="https://user-images.githubusercontent.com/919583/68900119-c2ea1c00-06f8-11ea-99b2-1eed010c88a6.png">

### Major updates

* Using a Google Spreadsheet with Geocoded address to power a CARTO table driving the map
* Code is based on the Probation Community Resources map, a Jekyll & Javascript project. Though is a major rewrite
* Followed the pattern of the Fusion Tables Map Template and split out customizations into separate files. cartodb_lib.js contains the majority of the code, and does not need to be edited beyond adding new filters
* All custom templates for hover, popup and table/list views are separated out into ejs templates
* Upgraded to Bootstrap 4
* Upgraded to Jquery 3.3.1
* Hosting on Netlify with HTTPS support

### Testing instructions

* Open up the old site ( http://macoupindirectory.com/) and compare functionality with the deploy preview for this PR (https://deploy-preview-1--macoupin-directory.netlify.com/). It should be identical
* Switch to mobile view and ensure that the layout and functionality works as expected
